Teapot/Teapot.swift

```diff
@@ -38,10 +38,18 @@ open class Teapot {
 
     open var logger = Logger()
 
+    let deliveryQueue: DispatchQueue
+
     // MARK: - Initialiser
 
-    public init(baseURL: URL) {
+    /// Initialise a Teapot instance.
+    ///
+    /// - Parameters:
+    ///   - baseURL: the base URL against which request paths are evaluated.
+    ///   - deliveryQueue: Queue where callbacks are executed on. Defaults to the main queue.
+    public init(baseURL: URL, deliveryQueue: DispatchQueue = .main) {
```

Comment:
One other thought: There may be some cases where you want particular calls to come back on a different queue, not all calls to the same queue. Thinking particularly of when we have to do fetch timestamp -> do a call with timestamp type stuff

Reply:
Based on the changes maybe rename this `defaultDeliveryQueue` instead of `deliveryQueue` - that way it's easier to tell by looking what's default vs passed in

Reply:
Great point!